rotkehlchen/db/search_assets.py

```diff
@@ -52,51 +52,53 @@ def _search_only_assets_levenstein(
     resolved_eth = A_ETH.resolve_to_crypto_asset()
     globaldb = GlobalDBHandler()
     treat_eth2_as_eth = db.get_settings(cursor).treat_eth2_as_eth
-    cursor.execute(
-        f'ATTACH DATABASE "{globaldb.filepath()!s}" AS globaldb KEY "";',
-    )
-    try:
-        query, bindings = filter_query.prepare('assets')
-        query = ALL_ASSETS_TABLES_QUERY.format(dbprefix='globaldb.') + query
-        cursor.execute(query, bindings)
-        found_eth = False
-        for entry in cursor:
-            lev_dist_min = 100
-            if entry[1] is not None:
-                lev_dist_min = min(
-                    lev_dist_min,
-                    levenshtein(filter_query.substring_search, entry[1].casefold()),
-                )
-            if entry[2] is not None:
-                lev_dist_min = min(
-                    lev_dist_min,
-                    levenshtein(filter_query.substring_search, entry[2].casefold()),
-                )
-            if treat_eth2_as_eth is True and entry[0] in (A_ETH.identifier, A_ETH2.identifier):  # noqa: E501
-                if found_eth is False:
-                    search_result.append((lev_dist_min, {
-                        'identifier': resolved_eth.identifier,
-                        'name': resolved_eth.name,
-                        'symbol': resolved_eth.symbol,
-                        'asset_type': AssetType.OWN_CHAIN.serialize(),
-                    }))
-                    found_eth = True
-                continue
+    with db.conn.critical_section():  # needed due to ATTACH. Must not context switch out of this
+        cursor.execute(
+            f'ATTACH DATABASE "{globaldb.filepath()!s}" AS globaldb KEY "";',
```

Comment:
is the key needed here? since the globaldb is not encrypted

Reply:
Absolutely yes. As this is an sqlcipher connection and we are attaching a normal sqlite DB to it. https://stackoverflow.com/questions/19742498/unable-to-attach-with-sqlcipher-encrypted-database-to-another-one